test/integration/query-interface.test.js

```diff
@@ -41,6 +41,26 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
     });
   });
 
+  describe('showAllTables', () => {
+    it('should not contain views', function() {
+      const cleanup = () => {
+        return this.sequelize.query('DROP VIEW V_Fail').reflect();
+      };
+      return this.queryInterface
+        .createTable('my_test_table', { name: DataTypes.STRING })
+        .tap(cleanup)
+        .then(() => this.sequelize.query('CREATE VIEW V_Fail AS SELECT 1 Id'))
+        .then(() => this.queryInterface.showAllTables())
+        .tap(cleanup)
+        .then(tableNames => {
+          if (tableNames[0] && tableNames[0].tableName) {
+            tableNames = tableNames.map(v => v.tableName);
+          }
+          expect(tableNames).to.deep.equal(['my_test_table']);
```

Comment:
Actually, I take back what I said before for this part of the code. The fact that you named the variable `tableNames` in the `.then` confused me. Here is the new request:

* Rename the variable from `tableNames` to `tables`, since it is not an array of strings, as the name would suggest
* Remove the if
* Add the following expects

```js
expect(tables).to.be.an('array');
expect(tables).to.have.length(1);
expect(tables[0].tableName).to.equal('my_test_table');
```

Reply:
> Remove the if

I can't remove the if. The `showAllTables()` returns either an array of strings of an array of object with name+schema in different dialects. I do think the API should be normalized to always return names+schemas but that's not in the scope of this task.

Reply:
Ah! I didn't know that, sorry. Thanks for the explanation. I thought objects were always returned. Do you know which dialect(s) return only names?

Reply:
Just checked the code in master. mysql, postgres and sqlit return only table names without schemas. If you check the test for `renameTable` you'll see the same if.
Changing the behavior of showAllTables() should be considered a change that breaks the API.

Reply:
Just checked the code in master. mysql, postgres and sqlit return only table names without schemas. If you check the test for renameTable you'll see the same if.
Changing the behavior of showAllTables() should be considered a change that breaks the API.

Reply:
@joaoe Thanks! Yeah, breaks API, so it will have to wait for the next major version. I have opened a new issue to track this (#11451 11451).

Your PR is good as is! Thanks!